specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/applicationWhitelistings.json

```diff
@@ -0,0 +1,606 @@
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AppWhitelistingGroups"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Security/locations/{ascLocation}/applicationWhitelistings/{groupName}": {
+      "get": {
+        "description": "Gets an application control VM/server group.",
+        "operationId": "AdaptiveApplicationControls_Get",
+        "x-ms-examples": {
+          "Gets a configured application control VM/server group": {
+            "$ref": "./examples/ApplicationWhitelistings/GetApplicationWhitelistingsGroup_example.json"
+          }
+        },
+        "tags": [
+          "applicationWhitelistings"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/AscLocation"
+          },
+          {
+            "$ref": "#/parameters/GroupName"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AppWhitelistingGroup"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "applicationWhitelistings"
+        ],
+        "x-ms-examples": {
+          "Update an application control VM/server group by adding a new file": {
+            "$ref": "./examples/ApplicationWhitelistings/PutApplicationWhitelistings_example.json"
+          }
+        },
+        "description": "Update an application control VM/server group",
+        "operationId": "AdaptiveApplicationControls_Put",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/AscLocation"
+          },
+          {
+            "$ref": "#/parameters/GroupName"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/AppWhitelistingGroupDataPutDataBody"
```

Comment:
the request body and response body must be same. it helps in get put pipeline. Customer GETs the resource, makes some changes and then PUts using the same model that he got using GET. If models are different, it becomes harder to use. 
Any properties which cant be set in request but come back during response can be marked as readOnly:true.